x-pack/plugins/secops/public/components/timeline/body/column_headers/filter.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+import { pure } from 'recompose';
+import { OnFilterChange } from '../../events';
+import { ColumnHeader } from './column_header';
+import { TextFilter } from './text_filter/text_filter';
+
+interface Props {
+  header: ColumnHeader;
+  onFilterChange?: OnFilterChange;
+}
+
+/** Renders a header's filter, based on the `columnHeaderType` */
+export const Filter = pure<Props>(({ header, onFilterChange = fp.noop }) => {
+  switch (header.columnHeaderType) {
+    case 'not-filtered':
+      return null;
```

Comment:
This seems better suited next to `default` below rather than here? Still looks fine, I would just find it slightly more elegant next to the default at line 33 or maybe it is gone completely and let the default take over as just returning `null`?